doc/devel/document.rst

```diff
@@ -317,7 +317,10 @@ target is unambiguous you can simply leave them out:
 
   `.LineCollection`
 
-and the link still works: `.LineCollection`.
+and the link still works: `.LineCollection`. Note that you should typically include
+the leading dot, otherwise Sphinx will try to resolve the object only in the current
+module. See also the explanation at `Sphinx: Cross-referencing Python objects
```

Comment:
in a sphinx page you can set a "current module", if you don't use the `.` it looks for the name in that module, if you use the dot it will look anythere in your project.

I think this is clear enough as-is and removing this bit of the text makes this PR mostly content-less.

Reply:
Reworded to

> Note that you should typically include
the leading dot. It tells Sphinx to look for the given name in the whole project.